src/app/filter/filter.service.ts

```diff
@@ -6,14 +6,10 @@ import { Filters } from './filter.model';
   providedIn: 'root'
 })
 export class FilterService {
-  readonly values$: BehaviorSubject<Filters>;
-
-  constructor() {
-    this.values$ = new BehaviorSubject<Filters>({
-      currentPage: 1,
-      entriesPerPage: 20
-    });
-  }
+  readonly values$ = new BehaviorSubject<Filters>({
+    currentPage: 1,
+    entriesPerPage: 20
+  });
 
   setState(filter: Filters): void {
     this.values$.next({
```

Comment:
you could use the trick of the destructuring here creating 3 const variables, const { currentPage, entriesPerPage.. } etc.

Reply:
I want to pass the entire state of filters as a single subject, so I think this is the only way of doing it.
Can you explain what you would do?

Reply:
mhm, something like that?
```typescript
 const { currentPage, entriesPerPage, nameFilter } = filter;
  this.values$.next({
      currentPage,
      entriesPerPage,
      nameFilter
    });
```

Reply:
or.... you could destruct it from the function parameter IIRC
```typescript
setState({ currentPage, entriesPerPage, nameFilter }: Filters): void {
  this.values$.next({
      currentPage,
      entriesPerPage,
      nameFilter
    });
```

Reply:
> mhm, something like that?
> 
> ```ts
>  const { currentPage, entriesPerPage, nameFilter } = filter;
>   this.values$.next({
>       currentPage,
>       entriesPerPage,
>       nameFilter
>     });
> ```

This is actually just a waste of memory, the BehaviourSubject already keeps a state by itself!

Reply:
> or.... you could destruct it from the function parameter IIRC
> 
> ```ts
> setState({ currentPage, entriesPerPage, nameFilter }: Filters): void {
>   this.values$.next({
>       currentPage,
>       entriesPerPage,
>       nameFilter
>     });
> ```

Doesn't work xD

Reply:
mhm, unbelievable

Reply:
`it works on my machine` but, consider that initially I could not make `npm install`, it was not working, so I have to update everything to a 14.x version

<img width="1557" alt="image" src="https://github.com/Helias/BG-Queue-Abuser-Viewer/assets/519778/78421c84-13e4-4cfa-8367-543d653cb65d">